src/main/java/io/reactivex/internal/operators/single/SingleTakeUntil.java

```diff
@@ -85,12 +85,9 @@ public void onSubscribe(Disposable d) {
         public void onSuccess(T value) {
             other.dispose();
 
-            Disposable a = get();
+            Disposable a = getAndSet(DisposableHelper.DISPOSED);
```

Comment:
Should be called at most once anyway, so biasing towards the `a != DisposableHelper.DISPOSED` being true.